### PR TITLE
Cloud vision color

### PR DESCRIPTION
**Pull request by deebot10**
## Implemented Features:
-Process started for gathering a hash from the CloudAPI that returns color info about each picture.

## Roadblocks/Logs/Errors:
-

## Testing:
-100%

## Notes/For the future:
-Similar to the categories collected from each picture, how we should go about iterating through the hash and saving the info we want into what tables in the DB.

### CODE REVIEW

File: app/facades/get_art_facade.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+require 'base64'
```
Comment (kabcouwer) [inline app/facades/get_art_facade.rb:1]: do we need base64 anymore?

File: app/services/cloud_vision_service.rb [hunk 2]
```diff
@@ -1,16 +1,13 @@
 class CloudVisionService
   class << self
     def artwork(value)
-      response = conn(value).post("/v1/images:annotate")
+      response = Faraday.post("https://vision.googleapis.com/v1/images:annotate?key=#{ENV['key']}", "{'requests':[{'image':{'source':{'imageUri':\"#{value}\"}},'features':{'maxResults':5,'type':'LABEL_DETECTION'}}]}",{'Content-Type': 'application/json'})
```
Comment (kabcouwer) [inline app/services/cloud_vision_service.rb:4]: can we update this method be updated with:
Faraday.post(url) do |req|
   req.params[:key] = ENV['key']
   req.params['reqests'] =
and so on

File: app/services/cloud_vision_service.rb [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
     end
 
-    def conn(value)
-      Faraday.new(url: 'https://vision.googleapis.com') do |req|
-        req.params[:key] = ENV['key']
-        # req.body['requests'][0]['image']['content'] = value  
-        
-      end
+    def artwork_color(value) 
+      response = Faraday.post("https://vision.googleapis.com/v1/images:annotate?key=#{ENV['key']}", "{'requests':[{'image':{'source':{'imageUri':\"#{value}\"}},'features':{'maxResults':5,'type':'IMAGE_PROPERTIES'}}]}",{'Content-Type': 'application/json'})
```
Comment (kabcouwer) [inline app/services/cloud_vision_service.rb:9]: same here?

**Review by kabcouwer — APPROVED** [review]
Looks good! Nice work

**Review by marlitas — APPROVED** [review]
Looks good. Also agree with Kim's comments and feedback. Happy to merge in, once those are addressed.

File: spec/facades/get_art_facade_spec.rb [hunk 4]
```diff
@@ -1,29 +1,114 @@
...
+      size = 'large'
+
+      stub_1 = WebmockStubs.mock_art
+
+      stub_request(:get, "https://api.artsy.net/api/artworks?size=5").to_return(status: 200, body: stub_1, headers: {})
+      stub_request(:post, "https://api.artsy.net/api/tokens/xapp_token").to_return(status: 200, body: stub_1, headers: {})  
+      
+      stub_2 = WebmockStubs.mock_colors
+
+      stub_request(:post, "https://vision.googleapis.com/v1/images:annotate?key=#{ENV['key']}").to_return(status: 200, body: stub_2)
```
Comment (marlitas) [inline spec/facades/get_art_facade_spec.rb:108]: Okay sweet. Once we merge this in I'll delete the old file that has our API key published.

**Review by marlitas — APPROVED** [review]

**Pull request merged**